### ISSUE TITLE
Improve the parsing or make the caching dynamic/better

### ISSUE THREAD

**Issue by arxari-archive** [Arxari/Shofi#10]
Rofi is able to launch fast while also keeping the cache up-to-date (if I install a new app then launch Rofi it will not show that app anymore).

Shofi should be able to do this too.

**Comment by arxari-archive** [comment]
Instead of proceeding further with the caching (due to it's inefficiency and issues) I've optimized the code a lot in commit [f00c128](https://github.com/Arxari/Shofi/commit/f00c1289923279114195616b389496667f2e91a5). This also resulted in the caching mechanism getting removed because it's not needed.

If someone is interested, here is how I optimized it:

Originally the script basically spammed ```grep``` to get the info from each .desktop file, now I use ```sed``` which can do this process much more efficiently since I don't spawn a trillion subprocesses anymore.

Before I stored apps in separate arrays for names and commands (because I'm dumb) now it's just name:command.

I also improved the search, instead of using ```grep``` and ```awk``` I now use ```mapfile```. Similar case as with the transition from ```grep``` to ```sed```, just a better way to do things.
